frontend/packages/process-editor/test/EventListeners.test.ts

```diff
@@ -0,0 +1,179 @@
+import { EventListeners } from './EventListeners';
+
+describe('EventListeners', () => {
+  describe('add', () => {
+    it('Adds a listener to the given event', () => {
+      const eventListeners = new EventListeners<{ event: () => void }>();
+      const fun = jest.fn();
+      const eventName = 'event';
+
+      eventListeners.add(eventName, fun);
+      eventListeners.triggerEvent(eventName);
+
+      expect(fun).toHaveBeenCalledTimes(1);
+    });
+
+    it('Supports adding multiple listeners to the same event', () => {
+      const eventListeners = new EventListeners<{ event: () => void }>();
+      const fun1 = jest.fn();
+      const fun2 = jest.fn();
+      const eventName = 'event';
+
+      eventListeners.add(eventName, fun1);
+      eventListeners.add(eventName, fun2);
+      eventListeners.triggerEvent(eventName);
+
+      expect(fun1).toHaveBeenCalledTimes(1);
+      expect(fun2).toHaveBeenCalledTimes(1);
+    });
+
+    it('Supports adding listeners to multiple events', () => {
+      const eventListeners = new EventListeners<Record<'event1' | 'event2', () => void>>();
+      const event1Fun = jest.fn();
+      const event2Fun = jest.fn();
+      const event1Name = 'event1';
+      const event2Name = 'event2';
+
+      eventListeners.add(event1Name, event1Fun);
+      eventListeners.add(event2Name, event2Fun);
+      eventListeners.triggerEvent(event1Name);
+      eventListeners.triggerEvent(event2Name);
+
+      expect(event1Fun).toHaveBeenCalledTimes(1);
+      expect(event2Fun).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('remove', () => {
+    it('Removes the given function from the given event listener', () => {
+      const eventListeners = new EventListeners<{ event: () => void }>();
+      const fun = jest.fn();
+      const eventName = 'event';
+      eventListeners.add(eventName, fun);
+
+      eventListeners.remove(eventName, fun);
+      eventListeners.triggerEvent(eventName);
+
+      expect(fun).not.toHaveBeenCalled();
+    });
+
+    it('Does not remove other functions than the given one', () => {
+      const eventListeners = new EventListeners<
+        Record<'event.of.interest' | 'another.event', () => void>
+      >();
+      const funToRemove = jest.fn();
+      const funOnSameEvent = jest.fn();
+      const funOnAnotherEvent = jest.fn();
+      const eventOfInterestName = 'event.of.interest';
+      const anotherEventName = 'another.event';
+      eventListeners.add(eventOfInterestName, funToRemove);
+      eventListeners.add(eventOfInterestName, funOnSameEvent);
+      eventListeners.add(anotherEventName, funOnAnotherEvent);
+
+      eventListeners.remove(eventOfInterestName, funToRemove);
+      eventListeners.triggerEvent(eventOfInterestName);
+      eventListeners.triggerEvent(anotherEventName);
+
+      expect(funOnSameEvent).toHaveBeenCalled();
+      expect(funOnAnotherEvent).toHaveBeenCalled();
```

Comment:
```suggestion
      expect(funOnSameEvent).toHaveBeenCalledTimes(1);
      expect(funOnAnotherEvent).toHaveBeenCalledTimes(1);
```

Reply:
I agree that this would make the test more robust, but I have delibaretely chosen to exclude `times` here since that's out of scope of what this test should cover. It should only verify that the functions are not removed, so `toHaveBeenCalled()` is sufficient.